pre_commit/resources/pre-push-tmpl

```diff
@@ -9,11 +9,12 @@ do
                 # Check that the ancestor has at least one parent
                 git rev-list --max-parents=0 "$local_sha" | grep "$first_ancestor" > /dev/null
```

Comment:
Would it make more sense to change this to `git rev-parse --verify "$first_ancestor~1" >/dev/null` since `first_ancestor` is set to the sha of the first commit that is not in a remote branch. That way we can leave the `--all-files` case for when its a new repo being pushed to a remote for the first time.